pymesos/process.py

```diff
@@ -147,7 +147,12 @@ def abort(self):
 
     def stop(self):
         Process.abort(self)
-        self.join()
+        assert (threading.current_thread() != self.accept_t), "Can't call stop in accept_t thread."
+        logger.debug("join ioloop thread.")
+        self.accept_t.join()
+        if threading.current_thread() != self.delay_t:
```

Comment:
how do u ensure `delay_t` finishes before main thread, when u call `stop` in `statusUpdate`? it seems u need to call `join` in main thread too.